src/libkomwm.py

```diff
@@ -341,37 +336,65 @@ def dump_priorities(prio_range, path, maxzoom):
                     group_prio = p[1]
                     group = ''
+                    group_comment = '# '
+
                 cl = p[0][0]
                 object_id = p[0][1]
+                auto_dr_type = None
+                auto_comment = None
+                if len(p[0]) == 4:
+                    auto_dr_type = p[0][2]
+                    auto_comment = p[0][3]
 
                 line_drules = ''
                 other_drules = ''
                 if cl in visibilities:
-                    for dr_type in sorted(visibilities[cl].keys()):
-                        for oid in sorted(visibilities[cl][dr_type].keys()):
-                            dr_zoom = dr_type + oid if oid else dr_type
-                            dr_zoom += ' ' + prettify_zooms(visibilities[cl][dr_type][oid], maxzoom)
-                            # Drules matching this prio_range and object_id.
-                            if (oid == object_id and
-                                ((prio_range == PRIO_OVERLAYS and dr_type in ('icon', 'caption', 'shield', 'pathtext')) or
-                                 (prio_range in (PRIO_FG, PRIO_BG_TOP) and dr_type in ('line', 'area')) or
-                                 (prio_range == PRIO_BG_BY_SIZE and dr_type == 'area'))):
-                                    if line_drules:
-                                        line_drules += ' and '
-                                    line_drules += dr_zoom
+                    for dr_type_comment in sorted(visibilities[cl].keys(), key = lambda drt: dr_types_order.index(drt[0])):
+                        for oid in sorted(visibilities[cl][dr_type_comment].keys()):
+                            dr_type, dr_auto_comment = dr_type_comment
+                            dr_zoom = dr_type + oid
+                            if dr_auto_comment is not None:
+                                dr_zoom = f'{dr_zoom}({dr_auto_comment})'
+                            dr_zoom += ' ' + prettify_zooms(visibilities[cl][dr_type_comment][oid], maxzoom)
+                            # Drules matching this prio_range and object_id and
+                            # - an auto priority dr_type match or
+                            # - any other non-auto dr_type suitable
+                            is_auto_dr_match = dr_type == auto_dr_type and dr_auto_comment == auto_comment
+                            is_not_auto_dr = auto_dr_type is None and dr_auto_comment is None
+                            is_suitable_for_range = (
+                                (prio_range == PRIO_OVERLAYS and dr_type in ('icon', 'caption', 'pathtext', 'shield')) or
+                                (prio_range in (PRIO_FG, PRIO_BG_TOP) and dr_type in ('line', 'area')) or
+                                (prio_range == PRIO_BG_BY_SIZE and dr_type == 'area'))
+                            if oid == object_id and (is_auto_dr_match or is_not_auto_dr and is_suitable_for_range):
```

Comment:
Again, (X or Y and Z) is undefined for me ..
Put braces here.
BTW, how python interpret this?